include/PieceFactory.h

```diff
@@ -0,0 +1,28 @@
+#include <memory>
+#include <unordered_map>
+#include <functional>
+
+#include "Piece.h"
+#include "Rook.h"
+#include "Knight.h"
+#include "Bishop.h"
+#include "Queen.h"
+#include "King.h"
+#include "Pawn.h"
+
+class PieceFactory {
```

Comment:
Consider using a factory pattern: The PieceFactory class can be refactored to adhere to the factory pattern more effectively. Instead of using an unordered_map to store creators, you can utilize a more flexible approach such as a std::map or an abstract factory interface.

Reply:
I have implemented std::map instead of the former.